plume-models/src/blogs.rs

```diff
@@ -195,14 +195,10 @@ impl Blog {
         }
     }
 
-    pub fn find_by_host(c: &PlumeRocket, host: Host) -> Result<Blog> {
-        let from_db = blogs::table
+    pub fn find_by_host(c: &PlumeRocket, host: Host) -> QueryResult<Blog> {
+        blogs::table
             .filter(blogs::custom_domain.eq(host))
-            .limit(1)
-            .load::<Blog>(&*c.conn)?
-            .into_iter()
-            .next();
-        from_db.ok_or(Error::NotFound)
+            .first::<Blog>(&*c.conn)
```

Comment:
What about adding a `map_err(|_| Error::NotFound)` here, to keep `Result<Blog>` as the return type (with the advantages it has, like having the possibility to "throw" it to get an error page)?

Reply:
And also it would allow you to remove the `From<diesel::result::Error> for ErrorPage` impl.

Reply:
> What about adding a `map_err(|_| Error::NotFound)` here

or `map_err(Error::from)` ?

Reply:
seemed like a good deal to me to directly map a `diesel::result::Error` to an `ErrorPage`
otoh, i can also see how this would lead to the kind of error handling that PHP + MySQL are famous for